multiversx_sdk/core/transactions_factories/relayed_transactions_factory.py

```diff
@@ -72,6 +73,32 @@ def create_relayed_v2_transaction(self,
             options=inner_transaction.options
         )
 
+    def create_relayed_v3_transaction(self,
+                                      transaction: ITransaction,
+                                      relayer_address: IAddress) -> Transaction:
+        """Relayer address must be in the same shard with sender address."""
+        if transaction.data.decode().startswith("relayed"):
```

Comment:
Maybe let the protocol enforce this, without additional logic in the SDKs?

Reply:
Removed